src/operator/numpy/np_tri_op-inl.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2019 by Contributors
+ * \file np_tri_op-inl.h
+ * \brief Function definition of the tri op
+ */
+
+#ifndef MXNET_OPERATOR_NUMPY_NP_TRI_OP_INL_H_
+#define MXNET_OPERATOR_NUMPY_NP_TRI_OP_INL_H_
+
+#include <dmlc/parameter.h>
+#include <vector>
+#include <string>
+#include <algorithm>
+#include "../mxnet_op.h"
+#include "../operator_common.h"
+#include "../elemwise_op_common.h"
+
+namespace mxnet {
+namespace op {
+
+struct TriParam : public dmlc::Parameter<TriParam> {
+  nnvm::dim_t N;
+  dmlc::optional<nnvm::dim_t> M;
+  int k;
+  int dtype;
+  DMLC_DECLARE_PARAMETER(TriParam) {
+    DMLC_DECLARE_FIELD(N)
+      .describe("Number of rows in the array.");
+    DMLC_DECLARE_FIELD(M)
+      .set_default(dmlc::optional<nnvm::dim_t>())
+      .describe("Number of columns in the array. "
+                "By default, M is taken equal to N.");
+    DMLC_DECLARE_FIELD(k)
+      .set_default(0)
+      .describe("The sub-diagonal at and below which the array is filled. "
+                "k = 0 is the main diagonal, while k < 0 is below it, "
+                "and k > 0 is above. The default is 0.");
+    DMLC_DECLARE_FIELD(dtype)
+    MXNET_ADD_ALL_TYPES
+    .set_default(mshadow::kFloat32);
+  }
+
+  void SetAttrDict(std::unordered_map<std::string, std::string>* dict) {
+    std::ostringstream N_s, M_s, k_s, dtype_s;
+    N_s << N;
+    M_s << M;
+    k_s << k;
+    dtype_s << dtype;
+    (*dict)["N"] = N_s.str();
+    (*dict)["M"] = M_s.str();
+    (*dict)["k"] = k_s.str();
+    (*dict)["dtype"] = dtype_s.str();
```

Comment:
```
(*dict)["dtype"] = String2MXNetTypeWithBool(dtype);
```
`String2MXNetTypeWithBool` lies in `incubator-mxnet/src/api/operator/op_utils.h`